src/Utils/Strings.php

```diff
@@ -265,6 +268,9 @@ public static function capitalize(string $s): string
 	 */
 	public static function compare(string $left, string $right, int $len = null): bool
 	{
+		$left = \Normalizer::normalize($left, \Normalizer::FORM_D); // form NFD is faster
+		$right = \Normalizer::normalize($right, \Normalizer::FORM_D); // form NFD is faster
```

Comment:
Does it make sense to call Normalizer directly? Or would we prefer to call self::normalize(). This will also normalizer line ending and some other stuff. I'm not sure on which level should be normalization done.

Current behaviour (proposed by @dg) causes inconsistencies as it is possible to get into situation where:

```php
Strings::compare($a, $b); // FALSE
Strings::normalize($a) === Strings::normalize($b); // TRUE
```

Is this expected behaviour? It seems confusing to me.

Reply:
Is it possible to call it after substring() ?

Reply:
No. Substring operates on code units, normalize may change length in code units (imho).